### ISSUE TITLE
Advanced Filter and "Opt in Primary Email" not working

### ISSUE THREAD

**Issue by Kristianostman** [salesagility/SuiteCRM#5938]
#### Issue
I have added the search field "Opt in Primary Email" from admin to Advanced Search for Contact List View. I have also Confirmed Opt-in in email settings. The three options  (Not Opt In, Opt In, Confirmed Opt In) are visible and can be selected, but without effect. The selection made is neiter saved even if I choose to save as custom filter.

#### Expected Behavior
The list should be populated according to selection, e.g. Confirmed Opt In.

#### Actual Behavior
The list is not filtered at all according to selected options in field Opt in Primary Email.

#### Steps to Reproduce
1. Go to Contacts
2. Click Filter / Advanced Filter
3. Select on or more options from Opt in Primary Email (NOTE! This field is not per default visible in Advanced Search, it has to be added from Admin -> Studio Builder -> Contacts -> Layout -> Filters -> Advanced Filter.)
4. Click Search button

#### Context
I want e.g. to get a list with Confirmed Opt In contacts after sending out a Manual Confirm Opt In email due to GDPR. Now I have to go to DB interfac in order to get a list of Confirmed contacts.

#### Your Environment
* SuiteCRM Version used: 7.10.5
* Chrome Version  65.0.3325.181 (64-bit)):
* MySQL, PHP 7:

**Comment by PedroErnst** [comment]
Hello @Kristianostman 

Thanks for reporting this issue.

I cannot seem to reproduce this issue. This is the result of filtering contacts for confirmed opt-in:

![image](https://user-images.githubusercontent.com/34056696/40537598-3c196b70-6007-11e8-8775-f917eaedcf53.png)

As with your other issue I'd ask you to supply more details regarding your environment so we can investigate this issue further.

Thanks!

**Comment by Kristianostman** [comment]
Hi @PedroErnst,

I am running MariaDB 10.2.8, PHP 7.1.17, Apache 2.4.6, OS Rel 7.5
PHP is excecuted via PHP-FPM.

This is an upgrade instance, gone through all steps from Sugar CE Version 6.5.24 (Build 509) to current Suite CRM Version 7.10.5 - Sugar Version 6.5.25 (Build 344)

**Comment by Kristianostman** [comment]
@PedroErnst  Please find a screen cast from our production instance https://drive.google.com/file/d/1YB4uYrWIffVTSJWPxTtjJXitAlAbxTI_/view. Filters working with other search fields but not with Opt in Primary Email.

**Comment by PedroErnst** [comment]
Thanks for the details @Kristianostman 

Looking at the video it seems the query is not actually saved. 

The might be something in either the PHP logs or the SuiteCRM logs to indicate the cause of this.

**Comment by Kristianostman** [comment]
This is now solved. My issue came from custom module. I had in some previous versions of Sugar CE the K-Reporter tool installed. After upgrade to Suite CRM the custom folder was unchanged. Even if I have uninstalled all modules not needed in the Module Loader (K-Reporter), there were still two instances of K-reporter visible and they could not be removed from there. I manually deleted the two instances (rows in the datbase table "upgrade_history". After this still one line in file /public_html/custom/application/Ext/TableDictionary/tabledictionary.ext.php requested for /public_html/custom/metadata/kreport.snapshot.metadata.php. My solution was to remove everything inside custom folder, since I do not use any custom addon anymore. After this the filters are working as they shall! Thanks for your input here, hopefully my information is useful for other having similar problems.
